### PR TITLE
Added SSL header correction functionality into plugin.

### PR DESCRIPTION
**Pull request by IcyApril**
Allows Flexible SSL to work transparently.

### CODE REVIEW

**Comment by jwineman** [comment]
I approve.  @thellimist feel free to merge if you're okay.

**Comment by thellimist** [comment]
@IcyApril 
also could you do 
`composer install`
`composer format`

Thank you

File: cloudflare.php [hunk 1]
```diff
@@ -468,6 +469,16 @@ function purgeCache()
...
 }
 
+function sslRewrite() {
+    if ( isset( $_SERVER['HTTP_X_FORWARDED_PROTO'] ) && $_SERVER['HTTP_X_FORWARDED_PROTO'] == 'https' ) {
+        $_SERVER['HTTPS'] = 'on';
+
+        return true;
+    }
+
+    return false;
```
Comment (jwineman) [inline cloudflare.php]: They're useful if we ever wrap the call in an if statement
```
if(sslRewrite()) {
//do something if we rewrote the header
}
```

**Comment by jwineman** [comment]
```
vendor/bin/phpcbf -n --standard=PSR2 --extensions=php,live.php src/
```
per https://github.com/squizlabs/PHP_CodeSniffer/wiki/Fixing-Errors-Automatically#using-the-php-code-beautifier-and-fixer

**Comment by jwineman** [comment]
basically `phpcbf` instead of `phpcs` for whatever command you run.

File: cloudflare.php (hunk 1 above)
Comment (jwineman) [inline cloudflare.php]: Also if we were to test `sslRewrite()` it would need to return `true` or `false` so we could `assert` whether it exited correctly.

Comment (thellimist) [inline cloudflare.php]: We can not write tests for sslRewrite if it's in `cloudflare.php`. It'll need to be moved inside `src/WordPress`.

Comment (jwineman) [inline cloudflare.php]: I know.  We'll figure that out at some point.  I only mention it to point out that we should be writing all our code in a way that is easy to test even if we can't actually write a test for it.

**Pull request merged**